mavros/src/plugins/sys_status.cpp

```diff
@@ -903,6 +912,19 @@ class SystemStatusPlugin : public plugin::PluginBase
 		}
 	}
 
+	/* -*- subscription callbacks -*- */
+
+	void statustext_cb(const mavros_msgs::StatusText::ConstPtr &req) {
+		mavlink::common::msg::STATUSTEXT statustext {};
+		ROS_WARN_COND_NAMED(req->text.length() >= statustext.text.size() - 1,
+				"sys", "Status text too long: truncating...");
+
+		statustext.severity = req->severity;
+		mavlink::set_string_z(statustext.text, req->text);
```

Comment:
Here we can add comment, that message defined "without null termination",
but quite often software treat it as null terminated.
And since it is not so limited like param name, better to keep 50-th character for terminator.

Reply:
not sure I understood this completely, but I added a comment